sqlglot/parser.py

```diff
@@ -383,7 +383,7 @@ class Parser:
         "offset": lambda self: self._parse_offset(),
     }
 
-    CREATABLES = {TokenType.TABLE, TokenType.VIEW, TokenType.FUNCTION}
+    CREATABLES = {TokenType.TABLE, TokenType.VIEW, TokenType.FUNCTION, TokenType.INDEX}
```

Comment:
you also need to add index to id_var so that you can do select 1 index -> select 1 as index

Reply:
fixed in #395